numpyro/infer/util.py

```diff
@@ -68,7 +68,7 @@ def log_density(model, model_args, model_kwargs, params, skip_dist_transforms=Fa
             scale = site['scale']
             # Early exit when all elements are masked
             if not_jax_tracer(mask) and mask is not None and not np.any(mask):
-                return jax.device_put(0.), model_trace
+                continue
```

Comment:
Thanks for fixing this. So if everything is masked this will return `0` as the log density?

Reply:
Yes. We start looping over the sites with `log_joint = 0.`